### PR TITLE
Create jspi-ot.md

### PR DESCRIPTION
**Pull request by fgmccabe**
A blog entry explaining that JSPI is going to origin trial and how to access the trial.

### CODE REVIEW

File: src/blog/jspi-ot.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+JSPI is an API that allows so-called sequential code – that has been compiled to WebAssembly – to access Web APIs that are _asynchronous_. Many Web APIs are crafted in terms of JavaScript `Promise`s: instead of immediately performing the requested operation they return a `Promise` to do so. When the action is finally performed, the browser’s task runner invokes any callbacks with the Promise. JSPI hooks into this architecture to allow a WebAssembly application to be suspended when the `Promise` is returned and resumed when the `Promise` is resolved.
+
+You can find out more about JSPI and how to use it [here](https://v8.dev/blog/jspi) and the specification itself is [here](https://github.com/WebAssembly/js-promise-integration).
+
+## Registering for the origin trial
+
+JSPI is still pre-release; it is going through a standardization process and will not be fully released until we get to phase 5 of that process. To use it today, you can set a flag in the Chrome browser; or, you can apply for an origin trial token that will allow your users to access it without having to set the flag themselves.
```
Comment (dtig) [inline src/blog/jspi-ot.md:17]: Phase 4 is more accurate here, we ship at Phase 4, and is merged into the official spec at Phase 5.

File: src/blog/jspi-ot.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+To register you can go [here](https://developer.chrome.com/origintrials/#/register_trial/1603844417297317889), make sure to follow the registration signup process. To find out more about origin trials in general, [this](https://developer.chrome.com/docs/web-platform/origin-trials) is a good starting place.
+
+## Some potential caveats
```
Comment (dtig) [inline src/blog/jspi-ot.md:21]: Add a section for where users should go to if they need find issues - links to the chromium bug tracker, emscripten issues, and the proposal issue tracker for API/Spec feedback.

File: src/blog/jspi-ot.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+There have been some [discussions](https://github.com/WebAssembly/js-promise-integration/issues) in the WebAssembly community about some aspects of the JSPI API. As a result, there are some changes anticipated, which will take time to fully work their way through the system. These do not affect the fundamental character of JSPI but, if you intend to use JSPI, you are advised to keep track of this.
```
Comment (dtig) [inline src/blog/jspi-ot.md:23]: If the expectation is that users will track this, please link the tracking bugs below.

File: src/blog/jspi-ot.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+title: 'WebAssembly JSPI is going to origin trial'
+description: 'We explain the start of the origin trial for JSPI'
+author: 'Francis McCabe, Thibaud Michaud, Ilya Rezvov, Brendan Dahl'
+date: 2024-03-06
+tags:
+  - WebAssembly
+---
+WebAssembly’s JavaScript Promise Integration (JSPI) API is entering an origin trial, with Chrome release M123. What that means is that you can test whether you and your users can benefit from this new API.
```
[the remark this answers is not in the document]
Comment (fgmccabe) [reply src/blog/jspi-ot.md:9]: Wrt Emscripten, this hasn't changed. As for flags, I am not sure what you are referring to.

[the remark this answers is not in the document]
Comment (dtig) [reply src/blog/jspi-ot.md:9]: It hasn't changed, but adding the emscripten release where there is stable JSPI support is helpful - especially when we're making backwards incompatible changes in the future. cc @brendandahl.

File: src/blog/jspi-ot.md (hunk 3 above)
Comment (fgmccabe) [reply src/blog/jspi-ot.md:23]: At the moment, the plan of action is to maintain the existing API until the end of the OT. We may introduce the new API while it is ongoing, but in any case, it is going to be near the end of the OT.

Comment (dtig) [reply src/blog/jspi-ot.md:23]: Did you mean to reply to a different comment? This one is for linking tracking issues (on crbug) for growable stacks and performance of wrapped calls.

File: src/blog/jspi-ot.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (brendandahl) [reply src/blog/jspi-ot.md:9]: The Emscripten compatible version would be 3.1.47 which includes the changes to type reflection API.

[the remark this answers is not in the document]
Comment (fgmccabe) [reply src/blog/jspi-ot.md:9]: done

File: src/blog/jspi-ot.md (hunk 2 above)
Comment (fgmccabe) [reply src/blog/jspi-ot.md:25]: done

**Review by dtig — APPROVED** [review]

**Pull request merged**